### PR TITLE
CR-1191 Changed ProductDTO to expose language to RHUI

### PR DESCRIPTION
**Pull request by devenrawat**
# Motivation and Context
Expose `language` as response to RHUI.
Use latest Product Reference library.
 

# What has changed
pom.xml change to use latest Product Reference.
ProductDTO changed to add language from ProductRefernece.

# How to test?
Run RHsvc and hit fulfilment endpoint, should get response with language e.g.
`
{
   "fulfilmentCode": "P_OR_H1",
   "productGroup": "QUESTIONNAIRE",
   "description": "Household Questionnaire for England",
   "caseTypes": [
      "HH",
      "SPG"
   ],
   "individual": false,
   "regions": [
      "E"
   ],
   "deliveryChannel": "POST",
   "handler": "QM",
   "language": "E"
}`

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1191

### CODE REVIEW

**Review by andrewjohnysons — APPROVED** [review]
Just checked with Ameet to see if we need to request the language but it appears that we filter this in RHUI - so I will approve

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/FulfilmentsServiceImplTest.java [hunk 1]
```diff
@@ -79,6 +79,7 @@ public void testFulfilmentsQuery() throws Exception {
     assertNull(capturedExampleProduct.getFulfilmentCode());
     assertNull(capturedExampleProduct.getDescription());
     assertNull(capturedExampleProduct.getHandler());
+    assertNull(capturedExampleProduct.getLanguage());
```
Comment (rpcatling) [inline src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/FulfilmentsServiceImplTest.java:82]: This is a poor test. Can we at least set the product to have a valid language, and assert that it is the correct one here.

**Review by rpcatling — CHANGES_REQUESTED** [review]
The unit test for the service needs improvement in the language test.

We also need to strengthen the testing in this area for the endpoint. Sadly you have inherited a week endpoint unit testing class for the /fulfilments.

At present I don't think we don't have any tests that prove that ProductDTO will serialise correctly to JSON through the endpoint.

The endpoint testing is inadequate , and we should add one or more tests to the FulfilmentsEndpointUnitTest .

This needs to follow our pattern of JSON based fixtures. You will need to add a JSON into the resources area following the pattern with a ProductDTO json , including the Language element.  Then read it in the unit test, mock it and check that the endpoint behaves properly.
Look at the other endpoint test classes for examples.

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/FulfilmentsServiceImplTest.java (hunk 1 above)
Comment (devenrawat) [reply src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/FulfilmentsServiceImplTest.java:82]: As per the comment in the code block 
` // Verify that nothing else was specified in the product search`
We are checking nothing else is set in  the product serach.

Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/FulfilmentsServiceImplTest.java:85]: OK fair enough, just needed to add language in the mock , so its a better test.

**Review by rpcatling — APPROVED** [review]

**Pull request merged**